File: lib/docker/config-templates.py
# Template para configuración segura de variables de entorno
# Este archivo se puede usar como base para generar configuraciones

import os
from typing import Optional, Dict, Any
from dataclasses import dataclass

@dataclass
class RobleConfig:
    """Configuración centralizada para servicios Roble"""
    base_url: str
    contract: str
    table_name: str
    user_email: Optional[str] = None
    user_password: Optional[str] = None
    token: Optional[str] = None
    mode: str = 'current'
    
    @classmethod
    def from_env(cls) -> 'RobleConfig':
        """Crear configuración desde variables de entorno"""
        config = cls(
            base_url=os.getenv('ROBLE_BASE_HOST'),
            contract=os.getenv('ROBLE_CONTRACT'),
            table_name=os.getenv('TABLE_NAME'),
            user_email=os.getenv('ROBLE_USER_EMAIL'),
            user_password=os.getenv('ROBLE_USER_PASSWORD'),
            token=os.getenv('ROBLE_TOKEN'),
            mode=os.getenv('ROBLE_MODE', 'current')
        )
        
        # Validar configuración requerida
        config.validate()
        return config
    
    def validate(self):
        """Validar que las variables requeridas estén presentes"""
        required = {
            'base_url': self.base_url,
            'contract': self.contract,
            'table_name': self.table_name
        }
        
        missing = [key for key, value in required.items() if not value]
        if missing:
            raise ValueError(f"Variables de entorno requeridas faltantes: {missing}")
        
        # Validar autenticación
        if self.mode == 'current':
            if not self.user_email or not self.user_password:
                raise ValueError("ROBLE_USER_EMAIL y ROBLE_USER_PASSWORD son requeridos en modo 'current'")
        else:
            if not self.token and not (self.user_email and self.user_password):
                raise ValueError("ROBLE_TOKEN o credenciales son requeridos en modo 'different'")
    
    def to_dict(self) -> Dict[str, Any]:
        """Convertir a diccionario (sin credenciales sensibles)"""
        return {
            'base_url': self.base_url,
            'contract': self.contract,
            'table_name': self.table_name,
            'mode': self.mode,
            'has_credentials': bool(self.user_email and self.user_password),
            'has_token': bool(self.token)
        }

# Función helper para cargar configuración
def load_roble_config() -> RobleConfig:
    """Cargar configuración Roble con validación automática"""
    try:
        return RobleConfig.from_env()
    except ValueError as e:
        print(f"❌ Error de configuración: {e}")
        print("💡 Asegúrate de tener las siguientes variables en .env.local:")
        print("   - ROBLE_BASE_HOST")
        print("   - ROBLE_CONTRACT") 
        print("   - TABLE_NAME")
        print("   - ROBLE_USER_EMAIL (modo current)")
        print("   - ROBLE_USER_PASSWORD (modo current)")
        raise

# Uso en el servicio
if __name__ == "__main__":
    try:
        config = load_roble_config()
        print(f"✅ Configuración cargada: {config.to_dict()}")
    except ValueError:
        exit(1)

File: lib/docker/config-templates.js
// Template para configuración segura de variables de entorno (JavaScript)
// Este archivo se puede usar como base para generar configuraciones

class RobleConfig {
    constructor() {
        this.baseUrl = process.env.ROBLE_BASE_HOST;
        this.contract = process.env.ROBLE_CONTRACT;
        this.tableName = process.env.TABLE_NAME;
        this.userEmail = process.env.ROBLE_USER_EMAIL;
        this.userPassword = process.env.ROBLE_USER_PASSWORD;
        this.token = process.env.ROBLE_TOKEN;
        this.mode = process.env.ROBLE_MODE || 'current';
        
        this.validate();
    }
    
    validate() {
        const required = {
            'baseUrl': this.baseUrl,
            'contract': this.contract,
            'tableName': this.tableName
        };
        
        const missing = Object.entries(required)
            .filter(([key, value]) => !value)
            .map(([key]) => key);
            
        if (missing.length > 0) {
            throw new Error(`Variables de entorno requeridas faltantes: ${missing.join(', ')}`);
        }
        
        // Validar autenticación
        if (this.mode === 'current') {
            if (!this.userEmail || !this.userPassword) {
                throw new Error("ROBLE_USER_EMAIL y ROBLE_USER_PASSWORD son requeridos en modo 'current'");
            }
        } else {
            if (!this.token && !(this.userEmail && this.userPassword)) {
                throw new Error("ROBLE_TOKEN o credenciales son requeridos en modo 'different'");
            }
        }
    }
    
    toObject() {
        return {
            baseUrl: this.baseUrl,
            contract: this.contract,
            tableName: this.tableName,
            mode: this.mode,
            hasCredentials: !!(this.userEmail && this.userPassword),
            hasToken: !!this.token
        };
    }
}

// Función helper para cargar configuración
function loadRobleConfig() {
    try {
        return new RobleConfig();
    } catch (error) {
        console.error(`❌ Error de configuración: ${error.message}`);
        console.log("💡 Asegúrate de tener las siguientes variables en .env:");
        console.log("   - ROBLE_BASE_HOST");
        console.log("   - ROBLE_CONTRACT");
        console.log("   - TABLE_NAME");
        console.log("   - ROBLE_USER_EMAIL (modo current)");
        console.log("   - ROBLE_USER_PASSWORD (modo current)");
        throw error;
    }
}

// Uso en el servicio
if (require.main === module) {
    try {
        const config = loadRobleConfig();
        console.log(`✅ Configuración cargada:`, config.toObject());
    } catch (error) {
        process.exit(1);
    }
}

module.exports = { RobleConfig, loadRobleConfig };
